modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/exec/exp/ExpressionFactoryImplTest.java

```diff
@@ -176,6 +189,60 @@ public void testRangeConditionWithNullIsIgnored() {
         assertEquals(List.of(new TestRange(new Object[]{1})), list);
     }
 
+    @ParameterizedTest(name = "condition satisfies: [{0}] the index")
+    @ValueSource(booleans = {true, false})
```

Comment:
I am not fully understand the idea behind `conditionSatisfyIdx = false` test case.
AFAIK the handling of this case was not changed by this patch. But maybe you know better why this test case is necessary.